composer/datasets/synthetic.py

```diff
@@ -17,40 +17,78 @@ class SyntheticDataType(StringEnum):
     SEPARABLE = "separable"
 
 
+class SyntheticDataLabelType(StringEnum):
+    CLASSIFICATION = "classification"
+    RANDOM_INT = "random_int"
+
+
 class MemoryFormat(StringEnum):
     CONTIGUOUS_FORMAT = "contiguous_format"
     CHANNELS_LAST = "channels_last"
     CHANNELS_LAST_3D = "channels_last_3d"
     PRESERVE_FORMAT = "preserve_format"
 
 
+'''
+    sample_pool_size -> batch size
+    batches_in_dataset (default 1)
+
+    X -> leave as-is
+
+    Y -> y_shape (default (1,))
+    y_type: Classification, Random
+    if classification - one_hot + num_classes
+    if random - shape
+'''
+
+
 class SyntheticDataset(torch.utils.data.Dataset):
 
     def __init__(self,
                  *,
-                 sample_pool_size: int,
-                 shape: Sequence[int],
-                 memory_format: Union[str, MemoryFormat],
-                 device: str,
-                 one_hot: bool,
-                 num_classes: int,
-                 transform: Optional[Callable] = None,
-                 data_type: SyntheticDataType = SyntheticDataType.GAUSSIAN):
-        self.size = sample_pool_size
-        self.shape = shape
-        self.input_data = None
-        self.input_target = None
-        self.memory_format = getattr(torch, MemoryFormat(memory_format).value)
-        self.device = device
+                 batch_size: int,
+                 data_shape: Sequence[int],
+                 batches_in_dataset: int = 1,
+                 data_type: SyntheticDataType = SyntheticDataType.GAUSSIAN,
+                 label_type: SyntheticDataLabelType = SyntheticDataLabelType.CLASSIFICATION,
+                 one_hot: Optional[bool] = None,
```

Comment:
Should `one_hot` be implied via the `label_type`? E.g. have `CLASSIFICATION_ONE_HOT`, CLASSIFICATION_INT`